src/erldns_resolver.erl

```diff
@@ -148,7 +148,7 @@ resolve_exact_type_match(Message, Qname, Qtype, Host, CnameChain, MatchedRecords
             false ->
               resolve_exact_type_match(Message, Qname, Qtype, Host, CnameChain, MatchedRecords, Zone, _AuthorityRecords, NSRecords)
           end;
-        {error, authority_not_found} ->
+        {error, _} ->
```

Comment:
Dialyzer doesn't like us explicitly wanting to crash. The other (theoretical) error case, based on the typespec:
```
 151: The pattern {'error', 'authority_not_found'} can never match the type {'error','no_question'} | {'ok',[any()]}
```
How should we handle `{error, no_question}`?